applications/GeoMechanicsApplication/custom_workflows/dgeosettlement.cpp

```diff
@@ -31,44 +31,60 @@ KratosGeoSettlement::KratosGeoSettlement()
     }
 }
 
-int KratosGeoSettlement::RunStage(const std::string&                      rWorkingDirectory,
-                                  const std::string&                      rProjectParametersFileName,
+int KratosGeoSettlement::RunStage(const std::filesystem::path&            rWorkingDirectory,
```

Comment:
Would it be doable to add a UT for this behavior?

Reply:
A legitimate question, but I'm afraid that the answer is that it will be rather difficult to add a unit test for this behavior. We might have a look together and investigate the possibilities that we see.

Reply:
As discussed, we'll move that to a separate PR, since it requires refactoring some of the implementation first.